pcdsdevices/attenuator.py

```diff
@@ -1071,6 +1072,114 @@ class AT2L0(FltMvInterface, PVPositionerPC, LightpathInOutMixin):
     # Summary for lightpath view
     num_in = Cpt(InternalSignal, kind='hinted')
     num_out = Cpt(InternalSignal, kind='hinted')
+    
+    """check for errors and return a string indicating any errors verbally"""
+    def _check_errors(signals: SignalToValue) ->  str:
+        errors = []
+        #sort out .motor from .motor.plc signals
+        for sig, value in signals.items():
+            if ("motor" in sig.name) ^ ("plc" in sig.name):
+                value=sig.metadata.get('severity').value
+
+            if value not in (0,""):
+                errors.append(f"{sig.name}: {value}")
+
+        if errors:
+            return "\n".join(["Error summary:"] + errors)
+
+        return "No errors"
+
+    """summarize the errors occurring at any time on any blade via a string"""
+    error_summary = Cpt(
+        MultiDerivedSignalRO,
+        calculate=_check_errors,
+        attrs=sum(
+            (
+                [
+                    f"blade_{_blade:02d}.state.error",
+                    f"blade_{_blade:02d}.state.error_id",
+                    f"blade_{_blade:02d}.state.error_message",
+                    f"blade_{_blade:02d}.motor.plc.err_code",
+                    f"blade_{_blade:02d}.motor.user_readback",
+                ]
+                for _blade in range(1,20)
+            ),
+            [],
+        ),
+    )
+
+    """check for errors and return an array of binaries 1=error, 0=no error"""
+    def _check_errors_bitmask(signals: SignalToValue):
+        errors = []
+        blade_errors=[]
+        for sig, value in signals.items():
+            if ("motor" in sig.name) ^ ("plc" in sig.name):
+                value=sig.metadata.get('severity').value
+            
+            if value not in (0,""):
+                blade_errors.append(1)
+            else:
+                 blade_errors.append(0)
+
+        #remove error messages for blade one
+        del blade_errors[0:5] 
```

Comment:
It's a bit weird (but not incorrect by any stretch!) to delete items from a list like this instead of just iterating over certain indices like:

```python
step = 5
start_index = 0
end_index = step

for _ in range(1, 19):
    error_count = sum(blade_errors[start_index:end_index])
    errors.append(1 if error_count >= 1 else 0)   # or bool(error_count) since False is 0 and True is 1
    start_index += step
    end_index += step
```

Reply:
right, I was torn on how to do that, it felt kind of wonky to be deleting from the front and letting the list re-index all the elements. I guess my reasoning at the time is that it would be slightly easier to check if the list was empty by the end to verify that the number of signals in the list returned zero when a mod 5 operator was applied. I also just wanted to go from blade 1 -> blade 2 -> etc. while compiling the error list just to make the logic a little easier in my head. With all that said, you're proposed solution makes sense and less wonky, I'll put it in!